packages/ui/src/primitives/tailwind/Input/index.tsx

```diff
@@ -45,6 +45,7 @@ export interface InputProps extends React.HTMLAttributes<HTMLInputElement> {
   errorBorder?: boolean
   maxLength?: number
   autoComplete?: string
+  pattern?: string
```

Comment:
Passing this prop is redundant since it's already included in `InputHTMLAttributes<HTMLInputElement>`.

Reply:
Actually you are right, pattern already is part of the InputHTMLAttributes interface, the thing is that we are not extending from that interface, so if I remove the definition it does complain.

I just spent a little bit of time analyzing and found that strangely, you guys have been setting prop types out of that interface, rather than extending, like **type** and **onChange**....

![image](https://github.com/user-attachments/assets/a25d7d2b-7158-464f-b4db-ed04b050eec7)

Is there a reason for this? shouldn't we simply also extend from InputHTMLAttributes like:

![image](https://github.com/user-attachments/assets/57182854-d37e-42cb-9f81-892692ec8625)

Once I do that the missing pattern definition disappears, I'm just trying to understand why you guys decided to use it at prop definition rather than by extending?

Regardless, I do agree that my definition of the property is redundant and incorrect, I just want to understand the logic behind setting types out of InputHTMLAttributes (Like type and onChange) rather than extending the interface.

Cheers!.

Reply:
The interface above is not on latest dev. Here's the latest interface: https://github.com/ir-engine/ir-engine/blob/af2902ba3af133c446de6e06aba1069d5da086f4/packages/ui/src/primitives/tailwind/Input/index.tsx#L37-L73

The reason className is omitted from the Input props is, we don't want to stick the `Input` design to Figma design everytime, without any overrides on its styles.